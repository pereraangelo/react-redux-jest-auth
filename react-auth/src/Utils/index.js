import moment from "moment";

export const getGreetingTime = () => {
  let currentTime = new Date();

  const splitAfternoon = 12; // 24hr time to split the afternoon
  const splitEvening = 15; // 24hr time to split the evening
  const currentHour = moment(currentTime).format("HH");

  if (currentHour >= splitAfternoon && currentHour <= splitEvening) {
    // Between 12 PM and 5PM
    return "Good Afternoon";
  } else if (currentHour >= splitEvening) {
    // Between 5PM and Midnight
    return "Good Evening";
  }
  // Between Dawn and noon
  return "Good Morning";
};



export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const simulateChangeOnInput = (wrapper, inputSelector, newValue) => {
  let input = findByTestAtrr(wrapper, inputSelector);
  input.simulate("change", {
    target: { name: inputSelector, value: newValue },
  });

  return findByTestAtrr(wrapper, inputSelector);
};


export const simulateCheckBox = (wrapper, inputSelector, newValue) => {
    let input = findByTestAtrr(wrapper, inputSelector);
    input.simulate('change', {target: {name:inputSelector,checked: newValue}});
  
    return findByTestAtrr(wrapper, inputSelector);
  };

  