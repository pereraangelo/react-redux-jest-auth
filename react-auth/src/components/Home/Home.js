import React from "react";
import { connect } from "react-redux";

export function Home(props) {
  const { loggingInProgress } = props;
  if (loggingInProgress) {
    return <div  data-test="loading">Loading...</div>;
  } else {
    return (
      <div className="homePage">
        <h1>Welcome to Synergy</h1>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    loggingInProgress: state.auth.loggingInProgress,
  };
};

export default connect(mapStateToProps)(Home);
