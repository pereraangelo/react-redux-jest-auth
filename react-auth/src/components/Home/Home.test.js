import React from "react";
import ReactDOM from 'react-dom'
import {shallow} from "enzyme"; 
import {Home} from "./Home";
import {MemoryRouter} from "react-router";
import { findByTestAtrr } from "../../Utils";
import renderer from 'react-test-renderer'

const defaultProps={
    logout: jest.fn(),
    isAuthenticated:false,
    userName:""
  }
  
  
  


describe('FAQ component',()=>{ 
    
    const wrapper = shallow(<Home {...defaultProps} />);
    it('Home commponent matches the snapshot',()=>{
        const tree=renderer.create(<MemoryRouter ><Home/></MemoryRouter>).toJSON()
        expect(tree).toMatchSnapshot()
    });
      
 
    it('Home page should load without crahing',()=>{
        const div = document.createElement('div');
        ReactDOM.render(<MemoryRouter ><Home/></MemoryRouter>, div);
    });

       
    it('show loading message until user logged in to the page',()=>{
        
        wrapper.setProps({ loggingInProgress: true });
        const loading=findByTestAtrr(wrapper, 'loading')
        expect(loading.length).toEqual(1)
 
    })
    

    

})

