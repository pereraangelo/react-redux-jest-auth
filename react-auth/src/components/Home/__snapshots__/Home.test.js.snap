// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ component Home commponent matches the snapshot 1`] = `
<div
  className="homePage"
>
  <h1>
    Welcome to Synergy
  </h1>
</div>
`;
